src/System.Text.Json/src/System/Text/Json/Serialization/ReflectionEmitMemberAccessor.cs

```diff
@@ -0,0 +1,152 @@
+            }
+
+            var dynamicMethod = new DynamicMethod(
+                ConstructorInfo.ConstructorName,
+                typeof(object),
+                Type.EmptyTypes,
+                typeof(ReflectionEmitMemberAccessor).Module,
+                skipVisibility: true);
+
+            ILGenerator generator = dynamicMethod.GetILGenerator();
+
+            if (realMethod == null)
+            {
+                LocalBuilder local = generator.DeclareLocal(type);
+
+                generator.Emit(OpCodes.Ldloca_S, local);
+                generator.Emit(OpCodes.Initobj, type);
+                generator.Emit(OpCodes.Ldloc, local);
+                generator.Emit(OpCodes.Box, type);
+            }
+            else
+            {
+                generator.Emit(OpCodes.Newobj, realMethod);
+            }
+
+            generator.Emit(OpCodes.Ret);
+
+            return (JsonClassInfo.ConstructorDelegate)dynamicMethod.CreateDelegate(typeof(JsonClassInfo.ConstructorDelegate));
+        }
+
+        public override object ImmutableCollectionCreateRange(Type constructingType, Type elementType)
+        {
+            MethodInfo createRange = ImmutableCollectionCreateRangeMethod(constructingType, elementType);
+
+            if (createRange == null)
+            {
+                return null;
+            }
+
+            return createRange.CreateDelegate(
+                typeof(DefaultImmutableConverter.ImmutableCreateRangeDelegate<>).MakeGenericType(elementType), null);
+        }
+
+        public override object ImmutableDictionaryCreateRange(Type constructingType, Type elementType)
+        {
+            MethodInfo createRange = ImmutableDictionaryCreateRangeMethod(constructingType, elementType);
+
+            if (createRange == null)
+            {
+                return null;
+            }
+
+            return createRange.CreateDelegate(
+                typeof(DefaultImmutableConverter.ImmutableDictCreateRangeDelegate<,>).MakeGenericType(typeof(string), elementType), null);
+        }
+
+        public override Func<object, TProperty> CreatePropertyGetter<TClass, TProperty>(PropertyInfo propertyInfo) =>
+            (Func<object, TProperty>)CreatePropertyGetter(propertyInfo, typeof(TClass));
+
+        private static Delegate CreatePropertyGetter(PropertyInfo propertyInfo, Type classType)
+        {
+            MethodInfo realMethod = propertyInfo.GetGetMethod();
+            Type objectType = typeof(object);
+
+            var dynamicMethod = new DynamicMethod(
+                realMethod.Name,
+                propertyInfo.PropertyType,
+                new[] { objectType },
+                typeof(ReflectionEmitMemberAccessor).Module,
+                skipVisibility: true);
+
+            ILGenerator generator = dynamicMethod.GetILGenerator();
+
+            generator.Emit(OpCodes.Ldarg_0);
+
+            if (classType.IsValueType)
+            {
+                generator.Emit(OpCodes.Unbox_Any, classType);
```

Comment:
Avoid boxing for perf?

Reply:
Note the previous code (before changed due to supporting struct) used a direct delegate which did not box on get\set.

Reply:
Yep, but to do that I need to multiply existing property info classes by two to support reference and value types. This will lead to unmaintable code which I'm trying to avoid. By the way `ReadStackFrame` stores a property value on the heap always, and only a json property info uses member accessors to write or to read some value. The property info is type safe. So I think that there is nothing to worry. The thing you are requesting can be addressed in the future, and at the same time we probably could optimize the stack frame type to store a value directly in it.

Reply:
ReadStackFrame only stored the "root" value that is going to be returned (`JsonSerialier.Parse<int>("1")`), not all property values on a POCO object for example.

I'm concerned that boxing on every stuct-based POCO property (int, DateTime, etc) and array item will affect perf, based on previous benchmarks from initial prototyping.